numpy/lib/function_base.py

```diff
@@ -4253,10 +4253,231 @@ def percentile(a, q, axis=None, out=None,
+    ``V`` is the value ``q`` of the way from the minimum to the
+    maximum in a sorted copy of ``V``. The values and distances of
+    the two nearest neighbors as well as the `interpolation` parameter
+    will determine the percentile if the normalized ranking does not
+    match the location of ``q`` exactly. This function is the same as
+    the median if ``q=0.5``, the same as the minimum if ``q=0`` and the
+    same as the maximum if ``q=1``.
+
+    Computation of weighted quantile uses normalized cumulative weights,
+    with some renormalization and edge-setting, to maintain consistency
+    with np.percentile.
+    The algorithm is illustrated here:
+
+    a = np.array([3, 5, 4])
+    q = [0.25, 0.5, 0.75]
+    axis = 0
+    weights = [1, 2, 3]
+
+    The np.percentile way, after expansion:
+
+    value       3   4   4   4   5   5
+    cum_prob    0  .2  .4  .6  .8   1
+    q=0.25            4
+    q=0.5                 4
+    q=0.75                    4.75
+
+    returns [4, 4, 4.75]
+
+    This method computes the cumulative probability bands associated with each
+    value, based on the weights:
+
+    value               3           4           5
+    weight              1           3           2  (total = 6)
+    cum. weight         .17         .67         1
+    prob_band         0 - .17   .33 - .67   .83 - 1  (.17-.33 and .67-.83 are
+                                                      transitions)
+    w slice             .17         .17         .17  (diff in upper bound,
+                                                      divided by weight)
+
+    lower_bound         0           .33         .83
+    upper_bound         .17         .67         1
+
+    # now substract the bound values by the left-most w-slice value
+
+    new_lower_bd        0           .17         .67
+    new_upper_bd        0           .5          .83  --> used to renormalize
+
+    renormalized_lower  0           .2          .8
+    renormalized_upper  0           .6          1
+
+    new line up         3       3       4       4       5       5
+    quantile bounds     0       0       .2      .6      .8      1
+
+    q=0.25                                 4
+    q=0.5                                     4
+    q=0.75                                          4.75
+
+    returns [4, 4, 4.75]
+
+    Examples
+    --------
+    >>> ar = np.arange(6).reshape(2,3)
+    >>> ar
+    array([[0, 1, 2],
+           [3, 4, 5]])
+    >>> np.quantile(ar, q=0.5, weights=np.ones(6).reshape(2,3))
+    2.5
+    >>> np.quantile(ar, q=0.5)
+    2.5
+    >>> np.quantile(ar, q=0.5, axis=0, weights=[1, 1])
+    array([ 1.5,  2.5,  3.5])
+    >>> np.quantile(ar, q=0.5, axis=1, weights=[1, 1, 1])
+    array([ 1.,  4.])
+    >>> np.quantile(ar, q=0.5, axis=1, weights=[0, 1, 2])
+    array([ 2.,  5.])
+    >>> np.quantile(ar, q=[0.25, 0.5, 0.75], axis=1, weights=[0, 1, 2])
+    array([[ 1.5,  4.5],
+           [ 2. ,  5. ],
+           [ 2. ,  5. ]])
```

Comment:
I don't like the fact this docstring is duplicated - I think one of these should be much smaller, as "Like <the other one>, but <something about 100>"